planetary-social/nos
# Re-apply Bech32 and TLV refactoring, this time with working note search

## Issues covered
#1294

Related:
#1319
#1334 
https://github.com/planetary-social/nos/pull/1319/files#r1688461214

## Description
This is a do-over of #1319 but also fixes the bug where pasting a `note...` into the search box didn't work. See 4b88f2fa3818b56854a9feea1149ba192b9f880d for the fix.

#1319 was reverted in #1334, so this is just reapplying that work, but this time without the note search bug.

## How to test
1. Navigate to Discover
2. Paste a `note...` into search, like `note1mvcpye0rerlfqraeyyafkahvvsxrr69kxhsly9rrehkevuwllf5szvrceq`
3. Observe the note
4. Also, make sure the rest of the app still works, especially around keys and npubs and notes.

## Screenshots/Video
https://github.com/user-attachments/assets/2bc4e421-3f89-45a5-aefd-2e65d830341b
